app/page/home/activity.tsx

```diff
@@ -51,181 +64,149 @@ const UserOpHistoryItem: React.FC<{
+  ) {
+    const link = `${explorerUrl}userOpHash/${txLog.receipt.userOpHash}?network=${chainName}`
     return (
-      <a
-        className="w-full flex items-center p-[14px_0px_14px_0px]"
-        href={`${explorerUrl}userOpHash/${userOpHash}?network=${chainName}`}
-        target="_blank">
-        {/* Status and to address */}
-        <div className="flex-grow flex flex-col items-start">
-          {/* Status and token amount */}
-          <div className="flex items-center mb-[4px]">
-            <OctagonXmark className="w-[16px] h-[16px] mr-[8px]" />
-            <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-              Failed
-            </div>
-            <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-              1.2
-            </div>
-            <div className="leading-[19px] text-[16px] text-[#000000]">ETH</div>
-          </div>
-          {/* To address */}
-          <div className="leading-[19px] text-[16px] text-[#bbbbbb] whitespace-nowrap">
-            to: 0x0000...00000
-          </div>
+      <div className="w-full flex items-center py-[13px] justify-between">
+        <div className="flex flex-col items-start gap-[8px]">
+          {/* Activity Topic element */}
+          {topic[topicType]}
+          {/* Activity Time element */}
+          <Time date={creationDate} time={creationTime} link={link} />
         </div>
-        {/* Transaction time */}
-        <div>
-          <div className="flex flex-col items-end">
-            <div className="mb-[4px] leading-[19px] text-[16px] text-[#000000]">
-              {creationDate}
-            </div>
-            <div className="leading-[19px] text-[16px] text-[#000000]">
-              {creationTime}
-            </div>
-          </div>
-        </div>
-      </a>
+        {/* Activity Status element */}
+        <Status status={status} tokenInfo={tokenInfo} />
+      </div>
     )
   }
 
-  if (txLog.status === TransactionStatus.Rejected) {
-    return (
-      <div className="w-full flex items-center p-[14px_0px_14px_0px]">
-        <div className="flex-grow flex flex-col items-start">
-          {/* Status and image */}
+  // Others' case
+  return (
+    <div className="w-full flex items-center py-[13px] justify-between">
+      <div className="flex flex-col items-start gap-[8px]">
+        {/* Activity Topic element */}
+        {topic[topicType]}
+        {/* Activity Time element */}
+        <Time date={creationDate} time={creationTime} />
+      </div>
+      {/* Activity Status element */}
+      <Status status={status} />
+    </div>
+  )
+}
+
+type TopicType = "send" | "contract" | "receive"
+
+const TopicTemplate: React.FC<{
+  Icon: React.ComponentType<{ className?: string }>
+  topic: string
+}> = ({ Icon, topic }) => {
+  return (
+    <div className="flex items-center">
+      <Icon className="w-[16px] h-[16px] mr-[8px]" />
+      <div className="text-[16px] text-[#000000]">{topic}</div>
+    </div>
```

Comment:
[nit]
It seems `<TopicTemplate/>` is about the UI template of the topic.

In addition to `<Icon/>` and **topic text**, we can observe all topics share the same view of: the wrapper(`LN95-96`), `<Time/>` and `<Status/>`. Maybe we can include them into `<TopicTemplate/>`.